validate/json_diff.py

```diff
@@ -173,6 +169,24 @@ def init_diff_logger(base_dir):
     logging.basicConfig(format="", handlers=[ch, fh], level=logging.DEBUG, force=True)
 
 
+def _get_primary_key(data, primary_key):
+    if isinstance(primary_key, str):
+        if primary_key == _HASH_PRIMARY_KEY:
+            return str(data)
+        else:
+            return data.get(primary_key, None)
+    elif isinstance(primary_key, list):
+        for primary_key in primary_key:
+            if primary_key in data:
+                return data[primary_key]
+    elif isinstance(primary_key, dict):
+        assert len(primary_key) == 1, f"Dict-type primary key should only contain one element."
```

Comment:
can we also log the primary_key in the case of failure for easier debugging?

Reply:
Done.